plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/OTelResourceProvider.java

```diff
@@ -32,6 +32,11 @@
  * This class encapsulates all OpenTelemetry related resources
  */
 public final class OTelResourceProvider {
+
+    private static final Object mutex = new Object();
+
```

Comment:
For [TelemetryTracerEnabledSanityIT Integ test we are using a multi node cluster with configuration min 2 data nodes. And node.java is initialised for all the nodes, hence the plugin is loaded too. So there are multiple calls to this.

Reply:
Why this is a problem? Each node would have own tracer, that's exactly how that should work in the wild.

Reply:
While running integ tests, All the nodes are initialised in one JVM, hence the node.java is initialised in single JVM for all the nodes and plugin is loaded multiple times.

[Here](https://github.com/opensearch-project/OpenSearch/blob/main/plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/OTelResourceProvider.java#L67) we are calling buildAndRegisterGlobal() which allows only one instance of OpenTelemetry to be created, otherwise throws exception. Hence we have added client side safety check to ensure only one call is made.

Reply:
Globals are pretty worrisome,  in any case I agree that we should not use the non-recommended functionality, the `OTelResourceProvider` could just do that instead (may be also log):

```
        OpenTelemetry openTelemetry = GlobalOpenTelemetry.get();
        if (openTelemetry != null) { 
            return openTelemetry;
        }
```

Reply:
Not sure If I understood your comment correctly, You want to put this check before setting the OpenTelemetry. But the problem here is [GlobalOpenTelemetry.get()](https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/main/java/io/opentelemetry/api/GlobalOpenTelemetry.java#L82) is calling set inside it and creating the instance everytime.

Reply:
And you are very right ... this is so bad .... OK, I think I know the way to do the OTEL style

Reply:
Unsurprisingly, there is `GlobalOpenTelemetry::resetForTest` - that should be addressing the testing issues. We may need to `OTelTelemetryPlugin` (in 1 test only) but at least we will keep mainline code clean.

Reply:
Ack,